### PR TITLE
chore: migrate AWS SDK for JavaScript v2 APIs to v3 in scripts folder

### PR DESCRIPTION
**Pull request by trivikr**
### Additional details

From AWS SDK for JavaScript v2 [README](https://github.com/aws/aws-sdk-js):
> We are formalizing our plans to make the Maintenance Announcement (Phase 2) for AWS SDK for JavaScript v2 in 2023.

This PR migrates AWS SDK for JavaScript v2 APIs to v3 using [aws-sdk-js-codemod](https://www.npmjs.com/package/aws-sdk-js-codemod).

```console
$ npx aws-sdk-js-codemod@0.26.1 -t v2-to-v3 scripts/binary/s3-api.ts
```

### Steps to test

N/A

### How has the user experience changed?

N/A

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by trivikr** [comment]
cc active contributors @emilyrohrbough @jennifer-shehane

File: package.json [hunk 1]
```diff
@@ -70,7 +70,8 @@
     "prepare": "husky install"
   },
   "devDependencies": {
-    "@aws-sdk/credential-providers": "3.53.0",
+    "@aws-sdk/client-s3": "3.441.0",
+    "@aws-sdk/credential-providers": "3.341.0",
```
Comment (emilyrohrbough) [inline package.json]: @trivikr why was `@aws-sdk/credential-providers` downgraded?

Comment (trivikr) [reply package.json]: My bad. It was a typo when manually adding the dependencies.
I added the latest using `yarn` in https://github.com/cypress-io/cypress/pull/28249/commits/c2d1d9fc9c6da87be8b1455f2131ff27f51c4301

File: package.json [hunk 2]
```diff
@@ -70,7 +70,8 @@
     "prepare": "husky install"
   },
   "devDependencies": {
-    "@aws-sdk/credential-providers": "3.53.0",
+    "@aws-sdk/client-s3": "3.450.0",
+    "@aws-sdk/credential-providers": "3.450.0",
```
Comment (emilyrohrbough) [inline package.json:74]: @trivikr this still seems to be a downgrade after you updated this with `yarn`.

Comment (trivikr) [reply package.json:74]: > this still seems to be a downgrade after you updated this with `yarn`.

I didn't get you. The diff added latest exact versions of `@aws-sdk/client-s3` and `@aws-sdk/credential-providers` at the time of development. What's being downgraded?

Comment (emilyrohrbough) [reply package.json:74]: Nope it's me. I was seeing `53` -> `45` not -> `450`. Sorry... 😅

**Review by emilyrohrbough — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -56,7 +56,7 @@ jobs:
           name: Download .circleci/workflows.yaml
           command: |
             if [[ "$CIRCLE_BRANCH" == "pull/"* ]]; then
-              curl -o workflows.yml https://raw.githubusercontent.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/develop/.circleci/workflows.yml
+              curl -o workflows.yml https://raw.githubusercontent.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/contributor-prs/.circleci/workflows.yml
```
Comment (emilyrohrbough) [inline .circleci/config.yml]:
```suggestion
              curl -o workflows.yml https://raw.githubusercontent.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/develop/.circleci/workflows.yml
```

**Review by mschile — APPROVED** [review]

**Comment by jordanpowell88** [comment]
We needed to revert this PR as it broke the CI job that uploads to S3 (see https://app.circleci.com/pipelines/github/cypress-io/cypress-publish-binary/3407/workflows/4f2225d3-8591-48d7-bb0e-d8a572e6490a/jobs/6045).

We should add this back in AND add appropriate tests to make sure this doesn't happen again.

**Pull request merged**